app/models/job_template.rb

```diff
@@ -65,30 +65,41 @@ def self.base_class
   # Import a template from ERB, with YAML metadata in the first comment.  It
   # will overwrite (sync) an existing template if options[:update] is true.
   def self.import(contents, options = {})
-    transaction do
-      metadata = parse_metadata(contents)
-      return if metadata.blank? || metadata.delete('kind') != 'job_template'
+    metadata = parse_metadata(contents)
+    import_parsed(metadata['name'], contents, metadata, options)
+  end
 
+  def self.import_raw!(contents, options = {})
+    template = import(contents, options)
+    template.save! if template
+    template
+  end
+
+  def self.import_parsed(name, text, metadata, options = {})
+    transaction do
+      metadata = metadata.dup
+      return if metadata.blank? || metadata.delete('kind') != 'job_template' ||
+                (metadata.key? 'model' && metadata.delete('model') != self.to_s)
+      metadata['name'] = name
       # Don't look for existing if we should always create a new template
-      existing = self.find_by_name(metadata['name']) unless options.delete(:build_new)
+      existing = self.find_by_name(name) unless options.delete(:build_new)
       # Don't update if the template already exists, unless we're told to
       return if !options.delete(:update) && existing
+      metadata.delete('associate')
```

Comment:
if this comes from foreman_templates, should it be possibly removed in `.import`?